### ISSUE TITLE
[Feature Request]: Improve linter settings

### ISSUE THREAD

**Issue by Gkrumbach07** [opendatahub-io/odh-dashboard#1713]
### Feature description

As features a growing there are a few things that could force better maintainability of functions and components

### No empty `children` when using a closing tag
_Example_
```tsx
❌     <div></div>

✅     <div>Hello World!</div>

✅     <div />
```

### Force a limit on optional function params
This is to limit the cases where we have to provide a bunch of undefined params in order to set the param at the end. This starts to become more relevant as we add features to existing features, and functions need to be adopted to accommodate them. 

The limit is TBD but options include:
- 1 optional param (this will completely avoid having to place undefined)
- 2-> _n_ optional params (a good number might be anything over 3 would fail)
_Example_
```tsx
❌ 
const myFunc = (
   a?: string,
   b?: string,
   c?: string,
   d?: string,
   e?: string,
) => { ... }
myFunc(undefined, undefined, undefined, undefined, "E")

❌ 
const myFunc = (
   a: string,
   b: string,
   c?: string,
   d?: string,
   e?: string,
) => { ... }
myFunc("A", "B", undefined, undefined, "E")

✅ 
const myFunc = (
   a: string,
   b: string,
   opt?: Partial<{
       c: string,
       d: string,
       e: string,
   }>,
) => { ... }
myFunc("A", "B", { e: "E" })

```

### Describe alternatives you've considered

these are not a required rule, and would add some extra dev time for new features as you would need to update all usage of a function, however they would greatly clean up the code.

**Comment by lucferbux** [comment]
Yes, we need to start adding more linting rules and enforcing them, we have room to improve in that field.

**Comment by adnankhan666** [comment]
https://issues.redhat.com/browse/RHOAIENG-513
